pantheon-bundle/frontend/src/app/productDetails.tsx

```diff
@@ -28,17 +28,17 @@ class ProductDetails extends Component<IProps, IState> {
         }
     }
 
-    public componentWillReceiveProps(nextProps) {
+    public componentDidUpdate() {
         // allow page load from productDetails to products listing
-        if (nextProps.productName !== undefined && nextProps.productName.trim() !== '') {
+        if (this.props.productName !== undefined && this.props.productName.trim() !== '') {
```

Comment:
@wesruv what is the thought behind this change? I believe this causes the user to be unable to actually navigate into product details page. I'm not sure why - I just noticed that clicking the 'product details' widget just takes you nowhere now.

Reply:
BLARG.

`componentWillReceiveProps ` is deprecated, I think it might have caused one of the test failures, but maybe I was mistaken. I'll try and revert it and rerun the tests.

See: https://github.com/redhataccess/pantheon/pull/354/files/b2e686dfa25a3ee42126782e0c3b18841ceae66e#diff-275ba574586207026ee0741897867bdbR117

It's definitely something we'll have to address at some point.

Reply:
Looks like it wasn't breaking the tests, but it does throw an error.

Reply:
Yeah, it's been awhile, but I vaguely recall being the person who wrote that and needing to specifically hunt down just the right method to make the logic work. Sucks that it's going away, I guess we'll cross that bridge when we come to it.